sunpy/coordinates/ephemeris.py

```diff
@@ -0,0 +1,243 @@
+    solar-system body at a specified time.
+
+    Parameters
+    ----------
+    body : `str`
+        The solar-system body for which to calculate positions
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~sunpy.coordinates.frames.HeliographicStonyhurst`
+        Location of the solar-system body in the `~sunpy.coordinates.HeliographicStonyhurst` frame
+    """
+    obstime = _astropy_time(time)
+
+    body_icrs = ICRS(get_body_barycentric(body, obstime))
+    body_hgs = body_icrs.transform_to(HGS(obstime=obstime))
+
+    return body_hgs
+
+
+def get_earth(time='now'):
+    """
+    Return a `~astropy.coordinates.SkyCoord` for the location of the Earth at a specified time in
+    the `~sunpy.coordinates.frames.HeliographicStonyhurst` frame.  The longitude will be 0 by definition.
+
+    Parameters
+    ----------
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~astropy.coordinates.SkyCoord`
+        Location of the Earth in the `~sunpy.coordinates.frames.HeliographicStonyhurst` frame
+    """
+    earth = get_body_heliographic_stonyhurst('earth', time=time)
+
+    # Explicitly set the longitude to 0
+    earth = SkyCoord(0*u.deg, earth.lat, earth.radius, frame=earth)
+
+    return earth
+
+
+def get_sun_B0(time='now'):
+    """
+    Return the B0 angle for the Sun at a specified time, which is the heliographic latitude of the
+    Sun-disk center as seen from Earth.  The range of B0 is +/-7.23 degrees.
+
+    Parameters
+    ----------
+    time : various
+        Time to use as `~astropy.time.Time` or in a parse_time-compatible format
+
+    Returns
+    -------
+    out : `~astropy.coordinates.Angle`
+        The position angle
+    """
+    return Angle(get_earth(time).lat)
+
+
+# Ignore warnings that result from going back in time to the first Carrington rotation
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore", ErfaWarning)
+
+    # Carrington rotation 1 starts late in the day on 1853 Nov 9
+    # according to Astronomical Algorithms (Meeus 1998, p.191)
+    _time_first_rotation = Time('1853-11-09 21:36')
+
+    # Longitude of Earth at Carrington rotation 1 in de-tilted HCRS (so that solar north pole is Z)
+    _lon_first_rotation = \
+        get_earth(_time_first_rotation).hcrs.cartesian.transform(_sun_detilt_matrix) \
+        .represent_as(SphericalRepresentation).lon.to('deg')
+
+
+def get_sun_L0(time='now'):
+    """
+    Return the L0 angle for the Sun at a specified time, which is the Carrington longitude of the
+    Sun-disk center as seen from Earth.
+
+    .. warning::
+       Due to apparent disagreement between published references, the returned L0 may be inaccurate
+       by up to ~2 arcmin, which translates in the worst case to a shift of ~0.5 arcsec in
```

Comment:
Actually, now that I think more about it, the issue is probably simply the exact start time of the first Carrington rotation.  I use the same time that Meeus uses (JD 2398167.4), hence the exact match isn't surprising, while the _Astronomical Almanac_ doesn't (explicitly) state what it uses.